src/diff_printer.rs

```diff
@@ -37,30 +41,39 @@ fn fmt_lines<F, D: Display>(lines: &str, mut current_line: Option<usize>,
 
 impl Display for DiffPrinter {
     fn fmt(&self, f: &mut Formatter) -> Result<(), Error> {
+        macro_rules! color {
```

Comment:
Can we move this into a function? I think it may be slightly cleaner as a higher order function that takes in a color and returns the relevant colorizer function

Reply:
If you want to as well, I think `Colorizer` may benefit from being its own type but I'll leave that up to your judgment on if that improves the code or makes it lengthier/hampers readability.

Reply:
I like the idea of `Colorizer` as separate type. Would make it simpler to add additional functionality in the future if desired.